main.go

```diff
@@ -117,12 +122,41 @@ func main() {
 			"the manager will watch and manage resources in all namespaces")
 	}
 
+	operatorInstallationNamespace := watchNamespace
```

Comment:
This entire section isn't clear to me.
RHOAM supported installation modes are as follows:
```
    - supported: false
      type: SingleNamespace
    - supported: false
      type: MultiNamespace
    - supported: false
      type: AllNamespaces
```
This means RHOAM can only run on it's installation namespace (and watch that namespace only). The exception seems to be if for some reason, you provide no watchnamespace, rhoam will attempt to run in cluster scoped mode.
      
Below what we are saying is following:
1. if watchns is "" - we run in cluster scope - that is fine
2. if watchns is not "" - we run in single ns target mode + operatorInstallationNamespace - so in the end, we are watching 2 namespaces - target ns and operator installation ns which is a multinamespace scoped installation

what we need here instead is to:
1. keep the old check as you do:
```
	watchNamespace, err := k8s.GetWatchNamespace()
	if err != nil {
		setupLog.Error(err, "unable to get WatchNamespace, "+
			"the manager will watch and manage resources in all namespaces")
	}
	``` 
	(although I don't like it..)
	2. If watchnamespace isn't "" -> add watchnamespace to Namespaces (only watchnamespace) without the ByObject part - unless I'm missing something?
	3. If watch namespace is "" -> empty managerCache.

Reply:
Done, hope as discussed. Thank you for detailed comment